File: Edu-Java/src/com/jetbrains/edu/java/JConfigurator.java
package com.jetbrains.edu.java;

import com.intellij.ide.fileTemplates.FileTemplateManager;
import com.intellij.lang.java.JavaLanguage;
import com.jetbrains.edu.jvm.gradle.GradleConfiguratorBase;
import com.jetbrains.edu.jvm.gradle.GradleCourseBuilderBase;
import com.jetbrains.edu.jvm.stepik.CodeTaskHelper;
import com.jetbrains.edu.learning.EduUtils;
import com.jetbrains.edu.learning.checker.TaskCheckerProvider;
import icons.EducationalCoreIcons;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class JConfigurator extends GradleConfiguratorBase {

  public static final String TEST_JAVA = "Tests.java";
  public static final String TASK_JAVA = "Task.java";
  public static final String MOCK_JAVA = "Mock.java";

  private final JCourseBuilder myCourseBuilder = new JCourseBuilder();

  @NotNull
  @Override
  public GradleCourseBuilderBase getCourseBuilder() {
    return myCourseBuilder;
  }

  @NotNull
  @Override
  public String getTestFileName() {
    return TEST_JAVA;
  }

  @Override
  public boolean isEnabled() {
    return !EduUtils.isAndroidStudio();
  }

  @NotNull
  @Override
  public TaskCheckerProvider getTaskCheckerProvider() {
    return new JTaskCheckerProvider();
  }

  @Override
  public String getMockFileName(@NotNull String text) {
    return CodeTaskHelper.fileName(JavaLanguage.INSTANCE, text);
  }

  @Override
  public String getMockTemplate() {
    return FileTemplateManager.getDefaultInstance().getInternalTemplate(MOCK_JAVA).getText();
  }

  @NotNull
  @Override
  public Icon getLogo() {
    return EducationalCoreIcons.JavaLogo;
  }
}

File: Edu-Java/src/com/jetbrains/edu/java/JConfigurator.kt
package com.jetbrains.edu.java

import com.intellij.lang.java.JavaLanguage
import com.intellij.openapi.project.Project
import com.jetbrains.edu.jvm.gradle.GradleConfiguratorBase
import com.jetbrains.edu.jvm.gradle.GradleCourseBuilderBase
import com.jetbrains.edu.jvm.gradle.checker.GradleTaskCheckerProvider
import com.jetbrains.edu.jvm.stepik.fileName
import com.jetbrains.edu.jvm.stepik.findCodeTaskFile
import com.jetbrains.edu.learning.EduUtils
import com.jetbrains.edu.learning.courseFormat.TaskFile
import com.jetbrains.edu.learning.courseFormat.tasks.Task
import com.jetbrains.edu.learning.courseGeneration.GeneratorUtils.getInternalTemplateText
import icons.EducationalCoreIcons
import javax.swing.Icon

class JConfigurator : GradleConfiguratorBase() {
  override val courseBuilder: GradleCourseBuilderBase
    get() = JCourseBuilder()

  override val testFileName: String
    get() = TEST_JAVA

  override val isEnabled: Boolean
    get() = !EduUtils.isAndroidStudio()

  override val taskCheckerProvider: GradleTaskCheckerProvider
    get() = JTaskCheckerProvider()

  override fun getMockFileName(text: String): String = fileName(JavaLanguage.INSTANCE, text)

  override fun getCodeTaskFile(project: Project, task: Task): TaskFile? {
    return findCodeTaskFile(project, task, taskCheckerProvider::mainClassForFile) ?: super.getCodeTaskFile(project, task)
  }

  override val mockTemplate: String
    get() = getInternalTemplateText(MOCK_JAVA)

  override val logo: Icon
    get() = EducationalCoreIcons.JavaLogo

  companion object {
    const val TEST_JAVA = "Tests.java"
    const val TASK_JAVA = "Task.java"
    const val MAIN_JAVA = "Main.java"
    const val MOCK_JAVA = "Mock.java"
  }
}